### PR TITLE
Adding torchtime

### PR DESCRIPTION
**Pull request by philipdarke**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by philipdarke** [comment]
@conda-forge/help-python, ready for review!

The package depends on `pytorch` which can be installed with `cpuonly` or a version of `cudatoolkit` (see Conda installation here https://pytorch.org/get-started/locally/). Any combination should satisfy the dependency. I think I have achieved this by testing under each CUDA version but I'd appreciate any feedback as this is my first conda-forge recipe.

Thanks very much!

**Comment by philipdarke** [comment]
@conda-forge-admin, please ping conda-forge/help-python

File: recipes/torchtime/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - python >=3.8,<3.10
+    - pip
+    - poetry >=1
+  run:
+    - python >=3.8,<3.10
+    - sktime
+    - requests
+    - scikit-learn
+    - tqdm
+    - pytorch  # pytorch is globally pinned
```
Comment (carterbox) [inline recipes/torchtime/meta.yaml:28]: You should include the constraints on the runtime dependencies except pytorch

https://github.com/philipdarke/torchtime/blob/7feb206557905282aba4b7bcbf5ab939794b7972/pyproject.toml#L13

**Review by carterbox — CHANGES_REQUESTED** [review]
This package contains python only. It doesn't need to be compiled. It is a noarch python package. Ignore fails on Windows if they are because of missing dependencies.

**Comment by philipdarke** [comment]
@conda-forge-admin, please ping conda-forge/help-python

Thanks Daniel - really appreciated.

The only difference to the main branch is now `torchtime/meta.yaml`. I've added the version constraints.

The recipe passes on Linux and Mac OS X, but fails on Windows with the message "Unsatisfiable dependencies for platform win-64: {MatchSpec("pytorch")}". As advised, I'm ignoring this.

Hopefully the recipe is good to go. Let me know if there is anything else.

File: conda-forge.yml [hunk 2]
```diff
@@ -2,4 +2,4 @@ channel_priority: strict
 azure:
   settings_win:
     pool:
-      vmImage: windows-2019
\ No newline at end of file
+      vmImage: windows-2019
```
[the remark this answers is not in the document]
Comment (carterbox) [reply conda-forge.yml:5]: This is not resolved. File changes still exist.

File: recipes/torchtime/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - pip
+    - poetry >=1
+  run:
+    - python >=3.8,<3.10
+    - sktime >=0.12.1,<1.13
+    - requests >=2.27.1,<2.28
+    - scikit-learn >=1.1.1,<1.2
+    - tqdm >=4.64.0,<4.65
+    - pytorch  # pytorch is globally pinned
+    - numpy >=1.21.0,<1.22
```
Comment (carterbox) [inline recipes/torchtime/meta.yaml:29]:
```suggestion
    - python >=3.8,<3.10
    - sktime >=0.12.1,<2
    - requests >=2.27.1,<3
    - scikit-learn >=1.1.1,<2
    - tqdm >=4.64.0,<5
    - pytorch >=1.11.0,<2
    - numpy >=1.21.0,<2
```
I was mistaken. Pytorch is actually only pinned for packages that link against it, but yours isn't compiled, so you can list the version compatibility. 

Also, this package poetry dependency spec is using caret not tilde.

https://github.com/philipdarke/torchtime/blob/7feb206557905282aba4b7bcbf5ab939794b7972/pyproject.toml#L13

https://python-poetry.org/docs/dependency-specification/

File: recipes/torchtime/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    - torchtime.constants
+    - torchtime.data
+    - torchtime.utils
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://philipdarke.com/torchtime
```
Comment (carterbox) [inline recipes/torchtime/meta.yaml:44]:
```suggestion
about:
  home: https://philipdarke.com/torchtime
  dev_url: https://github.com/philipdarke/torchtime
```

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/torchtime/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  noarch: python
+
+requirements:
+  host:
+    - python >=3.8,<3.10
+    - pip
+    - poetry >=1
+  run:
+    - python >=3.8,<3.10
+    - sktime >=0.12.1,<2
```
Comment (carterbox) [inline recipes/torchtime/meta.yaml:24]:
```suggestion
    - sktime >=0.12.1,<1
```

File: recipes/torchtime/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  noarch: python
+
+requirements:
+  host:
+    - python >=3.8,<3.10
+    - pip
+    - poetry >=1
+  run:
+    - python >=3.8,<3.10
+    - sktime >=0.12.1,<1
```
Comment (carterbox) [inline recipes/torchtime/meta.yaml:24]:
```suggestion
    - sktime >=0.12.1,<0.13.0
```
Carot is a complicated package spec

**Comment by philipdarke** [comment]
@conda-forge-admin, please ping conda-forge/help-python

Thanks for your work on this pull request - is there an action for me at this stage?

**Comment by carterbox** [comment]
Please note the extra constraint on sktime in the test section. This should no longer be necessary when the minimum sktime version is raised to 0.13.

**Pull request merged**